fastcrypto-vdf/src/vdf/wesolowski/fiat_shamir.rs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::math::hash_prime::{hash_prime, PrimalityCheck};
+use crate::vdf::wesolowski::WesolowskisVDF;
+use crate::{ParameterizedGroupElement, ToBytes, UnknownOrderGroupElement};
+use num_bigint::BigInt;
+use std::marker::PhantomData;
+
+pub trait FiatShamir<G: ParameterizedGroupElement + UnknownOrderGroupElement>: Sized {
+    /// Compute the prime modulus used in proving and verification. This is a Fiat-Shamir construction
+    /// to make the Wesolowski VDF non-interactive.
+    fn compute_challenge(vdf: &WesolowskisVDF<G, Self>, input: &G, output: &G) -> G::ScalarType;
+}
+
+impl<
+        G: ParameterizedGroupElement<ScalarType = BigInt> + UnknownOrderGroupElement,
+        const CHALLENGE_SIZE: usize,
+        P: PrimalityCheck,
+    > FiatShamir<G> for StrongFiatShamir<G, CHALLENGE_SIZE, P>
+{
+    fn compute_challenge(vdf: &WesolowskisVDF<G, Self>, input: &G, output: &G) -> BigInt {
+        let mut seed = vec![];
+
+        let input_bytes = input.to_bytes();
+        seed.extend_from_slice(&(input_bytes.len() as u64).to_be_bytes());
+        seed.extend_from_slice(&input_bytes);
+
+        let output_bytes = output.to_bytes();
+        seed.extend_from_slice(&(output_bytes.len() as u64).to_be_bytes());
+        seed.extend_from_slice(&output_bytes);
+
+        seed.extend_from_slice(&(vdf.iterations).to_be_bytes());
+        seed.extend_from_slice(&vdf.group_parameter.to_bytes());
+
```

Comment:
maybe replace the above with `bsc::to_bytes(&(vdf, input, output))` ?

Reply:
That would require implementing Serialize for quadratic forms which is not done yet, but we could do so. Deserialize is harder because the discriminant must be known, and I'm not sure it makes sense to just impl serialize. I'll think about it.